FLT/Mathlib/RingTheory/TensorProduct/Basis.lean

```diff
@@ -0,0 +1,51 @@
+import Mathlib.RingTheory.TensorProduct.Finite
+import Mathlib.LinearAlgebra.Basis.Basic
+import Mathlib.LinearAlgebra.TensorProduct.Pi
+import FLT.Mathlib.Algebra.Algebra.Tower
+
+section Basis
+
+open scoped TensorProduct
+
+attribute [local instance] Algebra.TensorProduct.rightAlgebra
+
+variable {R : Type*} (A : Type*) {B : Type*} {ι : Type*} [CommSemiring R]
+variable [CommSemiring A] [Algebra R A] [Fintype ι]
+variable [CommSemiring B] [Algebra R B]
+
+/-- The lift of an `R`-basis of `A` to a `B`-basis of the base change `A ⊗[R] B`. -/
+
+noncomputable
+def Basis.rightBaseChange (b : Basis ι R A) : Basis ι B (A ⊗[R] B) where
+  repr :=
+    letI : DecidableEq ι := Classical.typeDecidableEq ι
```

Comment:
I am not sure that this is right. There could be a competing decidable equality instance on \io if it is for example `Fin n`. I would ask on the Zulip if I were you (or I can ask if you'd rather). You could always put `[DecidableEq \io]` as a hypothesis, this is probably preferable.